src/ddmd/statementsem.d

```diff
@@ -3582,3 +3773,30 @@ Statement semanticScope(Statement s, Scope* sc, Statement sbreak, Statement scon
     scd.pop();
     return s;
 }
+
+static template TupleForeachArgs(bool isStatic, bool isDecl){
+    alias Seq(T...)=T;
+    static if(isStatic) alias T = Seq!(bool);
+    else alias T = Seq!();
+    static if(!isDecl) alias TupleForeachArgs = T;
+    else alias TupleForeachArgs = Seq!(Dsymbols*,T);
+}
+
+static template TupleForeachRet(bool isStatic, bool isDecl){
+    alias Seq(T...)=T;
+    static if(!isDecl) alias TupleForeachRet = Statement;
+    else alias TupleForeachRet = Dsymbols*;
+}
+
+TupleForeachRet!(isStatic,isDecl) makeTupleForeach(bool isStatic,bool isDecl)(Scope* sc,ForeachStatement fs,TupleForeachArgs!(isStatic,isDecl) args){
```

Comment:
or rather ... it should not be one.
templates are avoided in dmd.
I'd like to keep it that way.